code/modules/mob/mob_helpers.dm

```diff
@@ -530,11 +530,12 @@ GLOBAL_LIST_INIT(intents, list(INTENT_HELP,INTENT_DISARM,INTENT_GRAB,INTENT_HARM
 				var/mob/dead/observer/DM
 				if(istype(subject, /mob/dead/observer))
 					DM = subject
-				if(check_rights(R_ADMIN|R_MOD,0,M)) 							// What admins see
-					lname = "[keyname][(DM && DM.client && DM.client.prefs.toggles2 & PREFTOGGLE_2_ANONDCHAT) ? "*" : (DM ? "" : "^")] ([name])"
+				if(check_rights(R_ADMIN|R_MOD, FALSE, M)) 							// What admins see
+					lname = "[keyname][(DM?.client.prefs.toggles2 & PREFTOGGLE_2_ANON) ? (@"[ANON]") : (DM ? "" : "^")] ([name])"
+					//lname = "[keyname][(DM?.client.prefs.toggles2 & PREFTOGGLE_2_ANON) ? TRUE : (DM ? "" : "^")] ([name])"
 				else
-					if(DM && DM.client && DM.client.prefs.toggles2 & PREFTOGGLE_2_ANONDCHAT)	// If the person is actually observer they have the option to be anonymous
-						lname = "Ghost of [name]"
+					if(DM?.client.prefs.toggles2 & PREFTOGGLE_2_ANON)	// If the person is actually observer they have the option to be anonymous
+						lname = "<i>Anon</i> ([name])"
```

Comment:
I feel displaying `Anon` this prominently might get distracting.
What if we flip the format around a bit and make it `Ghost of John Nanotrasen (nanotrasenCEO)` instead?
Something like:

```dm
lname = name
if(DM)
	lname = "Ghost of [lname]"
	if(!(DM.client.prefs.toggles2 & PREFTOGGLE_2_ANON))
		lname = "[lname] ([keyname])"
```

Reply:
Thats just even more text, which will be a further distraction since that now impacts every other person's deadchat.